### PR TITLE
test: add dbt basic functional tests

### PR DESCRIPTION
**Pull request by BrechtDeVlieger**
### Description

Add the basic dbt test suite to the functional tests [as outlined in the docs](https://docs.getdbt.com/guides/dbt-ecosystem/adapter-development/4-testing-a-new-adapter). This should make it easier to validate that new changes don't break existing functionality.

Some of the tests are skipped, because they don't work (yet). The snapshot tests will be added in another PR. I'm not sure why the `TestBaseAdapterMethod` test fails. 

## Models used to test - Optional
/

## Checklist
- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You added unit testing when necessary
- [x] You added functional testing when necessary

### CODE REVIEW

**Review by Jrmyy — DISMISSED** [review]
Nice to see the first seeds for functional testing 🔥

File: tests/functional/adapter/test_basic_iceberg.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+iceberg_model_ephemeral = """
+  select
+      id,
+      name,
+      {{ cast_timestamp('some_date') }} as some_date
+  from {{ ref('ephemeral') }}
+"""
+
+
+def replace_single_model(model):
```
Comment (nicor88) [inline tests/functional/adapter/test_basic_iceberg.py:58]: nit: let's consider to rename this to something more meanifull e.g. `prepare_iceberg_models_config` or `get_iceberg_model_configs`

**Review by nicor88 — APPROVED** [review]
Amazing job, I left a small nit for renaming one of the method used to pick iceberg config.

**Comment by nicor88** [comment]
should close https://github.com/dbt-athena/dbt-athena/issues/142

File: tests/functional/adapter/test_basic_iceberg.py (hunk 1 above)
Comment (BrechtDeVlieger) [reply tests/functional/adapter/test_basic_iceberg.py:58]: Good point. Fixed it 👍

**Pull request merged**